keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,860 @@
+  - handling of successPolicy when all indexes succeeded
+  - handling of the `.spec.successPolicy.criteria.succeededIndexes` when some indexes remain pending
+  - handling of the `.spec.successPolicy.criteria.succeededCount` when some indexes remain pending 
+  - handling of successPolicy when some indexes of job with `backOffLimitPerIndex` fail
+
+##### e2e tests
+
+- Test scenarios:
+  - handling of successPolicy when all indexes succeeded
+  - handling of the `.spec.successPolicy.criteria.succeededIndexes` when some indexes remain pending
+  - handling of the `.spec.successPolicy.criteria.succeededCount` when some indexes remain pending
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind the `JobSuccessPolicy` feature gate.
+- Unit and integration tests passed as designed in [TestPlan](#test-plan).
+
+#### Beta
+
+- E2E tests passed as designed in [TestPlan](#test-plan).
+- Added a new reasons to the existing `job_finished_total` metric in [Monitoring Requirements](#monitoring-requirements).
+- Feature is enabled by default.
+- Address all issues reported by users.
+- Decided whether we introduce a `whenCriterionAchived=continue` and `whenCriterionAchived=continueWithRecreations`.
+Please see the [Possibility for the lingering pods to continue running after the job meets the successPolicy](#possibility-for-the-lingering-pods-to-continue-running-after-the-job-meets-the-successpolicy)
+section for discussion points.
+- Decided whether we introduce a new CronJob concurrentPolicy, `ForbidUntilJobSuccessful`.
+Please see the [Possibility for introducing a new CronJob concurrentPolicy, "ForbidUntilJobSuccessful"](#possibility-for-introducing-a-new-cronjob-concurrentpolicy-forbiduntiljobsuccessful)
+section for more details.
+
+#### GA
+
+- No negative feedback.
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade
+  - If the feature gate is enabled, `JobSuccessPolicy` are allowed to use only.
+  - If the feature gate is enabled without `JobSuccessPolicy`, 
+  the default values will be applied to a job object.
+  - Even if the feature gate is enabled, the Job controller doesn't update
+  `.status.conditions` in already finished jobs.
+- Downgrade
+  - Previously configured values will be ignored, and the job will be marked
+  as completed only when all indexes succeed.
+  - the Job controller doesn't update `.status.conditions` in already finished jobs.
+
+### Version Skew Strategy
+
+The apiserver's version should be consistent with the kube-controller-manager version,
+or this feature will not work.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: JobSuccessPolicy
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No.
```

Comment:
Can you be more specific why no? For example:

The default behavior of a job stays the same. The newly added field is optional and has to be explicitly set by the user to use this new feature.

Reply:
Done.